### PR TITLE
feat: use `@pinia/colada`

### PR DESCRIPTION
**Pull request by posva**
This PR simplifies the main todos code with cache invalidation thanks to [Pinia Colada](https://github.com/posva/pinia-colada) and adds a new page that implements Optimistic Updates so that the experience is 100% smooth even in a slow network

### CODE REVIEW

File: app/pages/todos.vue [hunk 1]
```diff
@@ -1,78 +1,112 @@
...
 
-async function addTodo() {
-  if (!newTodo.value.trim()) return
+const { data: todos } = useQuery({
+  key: ['todos'],
+  // NOTE: the cast sometimes avoids an "Excessive depth check" TS error
+  // using $fetch directly doesn't avoid the round trip to the server
+  // when doing SSR
+  // https://github.com/nuxt/nuxt/issues/24813
+  query: () => useRequestFetch()('/api/todos') as Promise<Todo[]>
```
Comment (atinux) [inline app/pages/todos.vue:20]:
```suggestion
  // Using useRquestFetch() toforward the cookies during SSR
  // https://github.com/nuxt/nuxt/issues/24813
  query: () => useRequestFetch()('/api/todos') as Promise<Todo[]>
```

File: app/pages/todos.vue [hunk 2]
```diff
@@ -1,78 +1,112 @@
...
+    // the first nextTick allows loading to become false and re enable the input
+    // the second nextTick allows the input to be rendered again so it can be focused
+    // a better solution would be to use a custom `v-focus` directive or a more elaborated focus management solution
+    nextTick()
+      .then(() => nextTick())
+      .then(() => {
+        console.log('loading', loading.value, !!newTodoInput.value?.input)
+        if (!newTodoInput.value?.input) {
+          console.error('Input not found')
+        }
```
Comment (atinux) [inline app/pages/todos.vue:52]: Do we need to keep this?

Comment (posva) [reply app/pages/todos.vue]: removed

**Review by atinux — APPROVED** [review]
❤️

**Pull request merged**